### PR TITLE
fix: Revert "fix: halt-height behavior is not deterministic (#305)"

### PR DESCRIPTION
**Pull request by JimLarson**
## Description

Refs Agoric/agoric-sdk#8223
Refs Agoric/agoric-sdk#8326

Revert #305 which causes test failure in Agoric/agoric-sdk.

### Author Checklist

I have...

* [x] included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
* [x] added `!` to the type prefix if API or client breaking change
* [x] targeted the correct branch (see [PR Targeting](https://github.com/cosmos/cosmos-sdk/blob/main/CONTRIBUTING.md#pr-targeting))
* [x] provided a link to the relevant issue or specification
* [x] followed the guidelines for [building modules](https://github.com/cosmos/cosmos-sdk/blob/main/docs/docs/build/building-modules/01-intro.md)
* [x] included the necessary unit and integration [tests](https://github.com/cosmos/cosmos-sdk/blob/main/CONTRIBUTING.md#testing)
* [x] added a changelog entry to `CHANGELOG.md`
* [x] included comments for [documenting Go code](https://blog.golang.org/godoc)
* [x] updated the relevant documentation or specification
* [x] reviewed "Files changed" and left comments if necessary
* [x] run `make lint` and `make test`
* [ ] confirmed all CI checks have passed

### Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items.*

I have...

* [ ] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
* [ ] confirmed `!` in the type prefix if API or client breaking change
* [ ] confirmed all author checklist items have been addressed 
* [ ] reviewed state machine logic
* [ ] reviewed API design and naming
* [ ] reviewed documentation is accurate
* [ ] reviewed tests and test coverage
* [ ] manually tested (if applicable)

### CODE REVIEW

**Comment by JimLarson** [comment]
Anticipating an immediate tag.

**Comment by michaelfig** [comment]
Weird golangci-lint failure.  Ugh.

**Review by michaelfig — APPROVED** [review]
LGTM!  :shipit:

File: CHANGELOG-Agoric.md [hunk 1]
```diff
@@ -45,4 +51,3 @@ Ref: https://keepachangelog.com/en/1.0.0/
 ### Bug Fixes
 
 * (snapshots) [#304](https://github.com/agoric-labs/cosmos-sdk/pull/304) raise the per snapshot item limit. Fixes [Agoric/agoric-sdk#8325](https://github.com/Agoric/agoric-sdk/issues/8325)
-* (baseapp) [#305](https://github.com/agoric-labs/cosmos-sdk/pull/305) Make sure we don't execute blocks beyond the halt height. Port of [cosmos/cosmos-sdk#16639](https://github.com/cosmos/cosmos-sdk/pull/16639)
```
Comment (mhofman) [inline CHANGELOG-Agoric.md]: We probably shouldn't delete this entry

Comment (JimLarson) [reply CHANGELOG-Agoric.md]: Restored.

**Pull request merged**